### DISCUSSION TITLE
[RFC] ConfigProvider support configuring all icon props

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#46889] [RFCs - archive]
## Motivation 

To customize icons used in components, like `closeIcon` of Tag

## API

In each component configuration in ConfigProvider

Use semantic name for each scenes of icon, .e.g `closeIcon` stands for icon to close something (Modal/Notification)

```tsx
<ConfigProvider icons={{ closeIcon: '' }}>
```

**Comment by li-jia-nan** [comment]
- #46894
- #47226
- #47200
- #47235
- #47244
- #47250
- #47252
- #47225
- #47274
- #47334
- #47301
- #47473
- #47561
- #47602
- #47589
- #47953

**Comment by li-jia-nan** [comment]


**Reply by li-jia-nan** [reply]
```ts
type IconType =
  | 'expandIcon' // Collapse、Menu
  | 'clearIcon' // Cascader、DatePicker、TimePicker、Input、Mentions、Select、TreeSelect
  | 'suffixIcon' // Cascader、Select、TreeSelect
  | 'feedbackIcon' // Form
  | 'upIcon' // InputNumber
  | 'downIcon' // InputNumber
  | 'removeIcon' // Cascader、Upload、Select、TreeSelect
  | 'selectionsIcon' // Transfer
  | 'allowClear' // Cascader

**Reply by MadCcc** [reply]
只有单个组件会用的可以不用放在这里面，后面支持组件 prop 就可以了。
`icon` 不用，这是每个组件的定制化内容，不具有语义。
`expandIcon` 在 Menu 和 Collapse 中使用的并不一致，暂时先不加。

```ts
type IconType =
  | 'closeIcon' // Alert、Drawer、FloatButton、Modal、Notification、Tag、Tour
  | 'clearIcon' // Cascader、DatePicker、TimePicker、Input、Mentions、Select、TreeSelect
  | 'suffixIcon' // Cascader、Select、TreeSelect 这个最好可以改个更具有语义的名字
  | 'removeIcon' // Cascader、Upload、Select、TreeSelect
```

**Comment by vitalijalbu** [comment]
is that supported? or not yet?

**Comment by Pareder** [comment]
Will it be possible to add Select component icons to `ConfigProvider` as it was stated in https://github.com/ant-design/ant-design/issues/54479?

**Comment by Pareder** [comment]
Also what do you think about adding DatePicker `suffixIcon` to `ConfigProvider`?
